marker_api/src/ast/item.rs

```diff
@@ -238,42 +240,182 @@ use impl_item_data;
 
 #[cfg(feature = "driver-api")]
 impl<'ast> CommonItemData<'ast> {
-    pub fn new(id: ItemId, span: SpanId, ident: Ident<'ast>) -> Self {
-        Self {
-            id,
-            span,
-            vis: Visibility::new(id),
-            ident,
-        }
+    // Here we can't `derive(TypedBuilder)`, as the created builder is private,
+    // due to the declaration of the item not being pub.
+    pub fn new(id: ItemId, span: SpanId, vis: Visibility<'ast>, ident: Ident<'ast>) -> Self {
```

Comment:
This surprised me, so I checked and the following works. You just need to make sure that `visibility::make(pub)` preceds the `TypedBuilder` derive due to the order of macro expansion.
```rust
#[repr(C)]
#[derive(Debug)]
#[cfg_attr(feature = "driver-api", visibility::make(pub))]
#[cfg_attr(feature = "driver-api", derive(typed_builder::TypedBuilder))]
struct CommonItemData<'ast> {
    id: ItemId,
    span: SpanId,
    vis: Visibility<'ast>,
    ident: Ident<'ast>,
}
```

Here, for example, it already works like that.
https://github.com/rust-marker/marker/blob/73835de0a82ebd33b959df200f66e5a3d7bf007a/marker_api/src/ast/stmt.rs#L59-L73